### ISSUE TITLE
ACF relationship field within repeater stuck in default loop

### ISSUE THREAD

**Issue by laras126** [timber/timber#362]
I'm using [ACF](http://advancedcustomfields.com) and [Timber](http://upstatement.com/timber/) and have a relationship field returning a post object within a repeater field. I want to get data from the related post to display in the template. The repeater field is called like this:

``` html+django
    {% for item in post.get_field('gear_recommendations') %}
        <div class="item">

            <h4>{{item.gear_title}}</h4>
            <img src="{{TimberImage(item.gear_photo).src}}" alt="{{item.title}}">
            <p>{{item.time_in_episode}}</p>

            {# Connect recommendation to a Deal post #}
            <p>Related deal: {{TimberPost(item.get_field('gear_to_deal')).title}}</p>

        </div>
    {% endfor %} 
```

This returns the title of the current post rather than the related one. I did it like this before with PHP which appears to be overriding the default loop somehow with $post:

``` php
    <?php
        $posts = get_field('related_meister');
        foreach( $posts as $post): // variable must be called $post
            ...
```

Using the debug bar, it looks like ACF is returning the post ID:

``` php
    [gear_recommendations_0_gear_to_deal] => Array
                                     (
                                            [0] => 106
                                        )
```

Should I be referencing that ID and getting the post data from there, rather than going directly through the field name? How do I do that? Or is there a way to get out of the default loop so I can get to the relationship data the way I am doing it now?

Help is much appreciated! And thanks so much for your hard work - Timber is so so so awesome.

**Comment by robertcrenteria** [comment]
@jarednova your solution almost works but for some reason it only returns the first item of the relationship items. I'm assuming it's because of the [0] but when I remove that it doesn't return anything. Any ideas? Thanks a bunch. My relationship field is a part of an acf options page.

`
{% for item in options.the_mountain %}
    <div class="submenu">
        <h4 class="black-text">{{item.menu_title}}</h4>
        <ol class="submenu-items">
	    {% for menu in item.menu_items %}
	        <li>
	        	{{  TimberPost( item.menu_items[0] ).title  }} 
	        </li>
	    {% endfor %}
	    </ol>
    </div>
{% endfor %}
`

**Comment by doifeellucky** [comment]
Hi

Having done a search, and looked at the docs, this seems to be similar to the problem I'm trying to solve when using ACF.

I've created an ACF 'related_post' set as a 'page link' and want to show the post title (for now) of the related post in the current post. 

I'm currently using the following which shows the full URL of the related post as I'd expect.

<p><a href="{{post.get_field('related_post')}}" target="_blank">{{post.get_field('related_post')}}</a></p>

How do I show the related post title?  I've different combinations, from the above and the docs but nothing works.

Thanks

**Comment by laras126** [comment]
@doifeellucky I've definitely made that mistake before! In your `href`, make sure you are referring to the link as well not the enture object. 

I like to set a little Twig variable to store the `post.get_field('related_post')` in for brevity - something like:

```
{% set related = post.get_field('related_post') %}
<p><a href="{{related.link}}" target="_blank">{{related.title}}</a></p>
```

**Comment by doifeellucky** [comment]
@laras126 Thanks Lara.  Yes I noticed that I needed to change it to `{{post.get_field('related_post').link}}` when I got an error.

I've been looking at your tacklebox series.  With Timber I generally tend to struggle the most with what goes in the php file, having used Expression Engine and Craft in the past where this isn't really required.  I generally get something to work, so far, via trial and error, but would prefer to have a more thorough understanding.  Do any of your videos cover this from a designers perspective?

Thanks
